### ISSUE TITLE
Performer Attention Masking Question

### ISSUE THREAD

**Issue by jdefriel** [google-research/google-research#1461]
I would like to use the performer in a masked language model and am looking for some insight into where to apply the attention mask since the order of operations is changed in the kernel and an explicit soft max isn't taken. Do I mask Q instead of (QK^T) since I take (KV) first?

**Comment by jdefriel** [comment]
Actually it makes more sense to mask data_dash and data_diag before they go through the exp function to remove influence of padding. Thoughts?

**Comment by jdefriel** [comment]
Found where the padding mask is applied to the values in performer/fast_attention/jax/attention_module.py line 121. It doesn't look like it carried over to the tensorflow version for anyone that has this same question.
